Gems/Atom/RPI/Code/Source/RPI.Public/Image/ImageSystem.cpp

```diff
@@ -200,6 +200,39 @@ namespace AZ
             return m_systemImages[static_cast<size_t>(simpleImage)];
         }
 
+        bool ImageSystem::RegisterAttachmentImage(const AttachmentImage& attachmentImage)
+        {
+            if (m_registeredAttachmentImages.find(attachmentImage.GetAttachmentId()) != m_registeredAttachmentImages.end())
+            {
+                AZ_Assert(false, "AttachmangeImage with name '%s' was already registered", attachmentImage.GetAttachmentId().GetCStr());
+                return false;
+            }
+
+            m_registeredAttachmentImages[attachmentImage.GetAttachmentId()] = &attachmentImage;
```

Comment:
If you save the iterator, you could update it without doing another find.

```suggestion
            auto iter = m_registeredAttachmentImages.find(attachmentImage.GetAttachmentId());
            if (iter != m_registeredAttachmentImages.end())
            {
                AZ_Assert(false, "AttachmangeImage with name '%s' was already registered", attachmentImage.GetAttachmentId().GetCStr());
                return false;
            }

            iter ->second = &attachmentImage;
```

Reply:
the iter would be invalid when run to the line 212
iter ->second = &attachmentImage;

Reply:
Oh yeah, sorry I was moving too fast.